### ISSUE TITLE
Compile error with OpenMP

### ISSUE THREAD

**Issue by Thyra** [ghzuo/CVTree#5]
Hi,
great work on making this software available! :-)
I'm getting an error running the `make` step which seems to come from an OpenMP instruction:

```
CVTree/cvtree/treemeth.cpp: In member function ‘void NeighborJoint::njnearest(const Mdist&, StarTree&, Neighbor&)’:
CVTree/cvtree/treemeth.cpp:106:27: error: expected ‘+’, ‘*’, ‘-’, ‘&’, ‘^’, ‘|’, ‘&&’, ||’, ‘min’ or ‘max’ before ‘mindd’
 #pragma omp for reduction(mindd : nb)
```

It does work if I compile without OpenMP (by just commenting out lines 42-46 in the CMakeLists.txt) but then CVTree cannot run in parallel, right? I'm trying to cluster about 500 whole plant genomes so performance is certainly a limiting factor.
I'm using `gcc` version 10.1.0 and OpenMP version 4.5

**Comment by Thyra** [comment]
In case anyone else is having the same problem: I removed all OpenMP #pragma instructions from the `treemeth.cpp` because it only concerns the neighbor joining (which I suspected doesn't take too much time anyway in my case) and it led to the same resulting trees while speeding up the process a lot. Not optimal but a workaround for now.

**Comment by ghzuo** [comment]
It seems that the problem was caused by OpenMP can not handle my declare of "mindd" at lines 84~86. However, as the manual of OpenMP, the function had been supported since OpenMP 2.1.6. The option is used for the parallel of NJ. And it has meaningful improvement only when the number of genomes is more than ~1000. 

Furthermore, can you tell me what operating system you use? I had tested the program on Linux (GCC 8 and 9) and macOS (clang 11,12 + OpenMP (installed by brew)). And as the manual of GCC,  it seems that the GCC included the OpenMP. And it included openmp5 since gcc9.  And you should use the OpenMP of GCC directly. And noted that the GCC command on macOS is a link to clang.

**Comment by Thyra** [comment]
Hey, thanks for looking into this! I'm intending to use it on one of our HPC servers and have tested compiling it on another HPC cluster as well to see if I can localise the problem and here is what I found:

The server where it doesn't work uses the following software:
```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

$ gcc --version
gcc (GCC) 10.1.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

On the other HPC cluster it does work though

```
$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux Server"
VERSION="7.9 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.9 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.9:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.9"

$ gcc --version
gcc (Spack GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

In both cases, I'm loading gcc through the `module` system. The default gcc on both systems is 4.8.5 with which expectedly it does not work. The second HPC (where 10.2.0 does work) also has a gcc 7.3.0 available and that worked, too.
So apparently either there's a bug in gcc 10.1 that got fixed in 10.2, or it has something to do with the underlying operating system, or something weird is going on with the module loading where OpenMP isn't loaded properly. Unfortunately I know almost nothing about C++ and how compilation works in general so I don't really know what else to look for. If you would like I can ask our IT department to look into this and see if they can figure out what is going on, otherwise feel free to close the issue as I have found a workaround that is sufficient for my case.
